### ISSUE TITLE
Range query does not support _name anymore in 2.x

### ISSUE THREAD

**Issue by SMUnlimited** [elastic/elasticsearch#15306]
QueryParsingException[[range] query does not support [_name]]
	at org.elasticsearch.index.query.RangeQueryParser.parse(RangeQueryParser.java:108)

Looking at RangeQueryParser the _name field has been removed between 1.7 and 2.0. Possibly related to #11744 ?

It appears to be fixed again in 'master' but may need to be fixed independently for the 2.x line.

Following query was used

````
{
  "nested": {
    "path": "inner_name_only",
    "score_mode": "avg",
    "query": {
      "range": {
      	"inner_name_only.inner_date_field": {
          "gte" : "1999/01/01",
          "lte" : "2020/01/01",
	   "_name" : "query1"
      	}
      }
    }
  }
}
````

**Comment by cbuescher** [comment]
Thanks for reporting this, to me it looks like the expected place for the "_name" parameter was changed once with a0af88e9 to be on the top level, so in 2.x the following should work:
```
"range": {
      "inner_name_only.inner_date_field": {
        "gte": "1999/01/01",
        "lte": "2020/01/01"
      },
      "_name": "query1"
    }
```
However, as you mentioned in #11974 support for "_name" was added again on the main level alongside e.g. "boost" on master, but this was not ported to the 2.x branch. I think we should also add this back again on 2.x since the original move to the top-level of the query seems incidental. @javanna wdyt?

**Comment by jpountz** [comment]
Agreed it's weird to have back compat on master that does not exist on 2.x.

**Comment by cbuescher** [comment]
@jpountz thanks, I just checked the situation on master again and will add the test if you agree. The version we seem to be deprecating on master is the only one allowed on 2.x, so I'd be in favour of changing that to the behaviour on master (allowing both places, but using ParseField to deprecate the use on the top level.

**Comment by cbuescher** [comment]
Closed by #15394
